what-is-chalk.mdx

```diff
@@ -18,215 +18,467 @@ import { PyDiffEditor } from '@/components/Editor'
     )
 ```
 
-### Resolvers
+Chalk queries allow you to explicitly express the features you want returned:
+
+```shell
+chalk query --in user.id=241 --out name_match --out transactions
+```
+
+```
+Results
+https://chalk.ai/environments/{environment}/query-runs/{query_id}
+Branch: elvis
+Environment: {environment}
+
+ Name                       Hit?  Value
+────────────────────────────────────────
+ user.name_match        41.0
 
-Next, tell Chalk how to compute your features.
-Chalk ingests data from your existing data stores,
-and lets you use Python to compute features with
-[feature resolvers](/docs/resolver-overview).
-Feature resolvers are declared with the decorators `@online` and
-`@offline`, and can depend on the outputs of other feature resolvers.
 
-Resolvers make it easy to rapidly integrate a wide variety of data
-sources, join them together, and use them in your model.
+user.txns
+
+ id    updated_at                          transaction_status  user_id  session_id  item_count  cheapest_line_item_price  most_expensively_valued_product_id  created_at
+─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+ 17    "2024-12-06T19:53:21.575073+00:00"  "failed"            27       16281       2           34                        20364                               "2024-12-06T00:36:07.348750+00:00"
+ 18    "2024-11-27T01:50:03.002378+00:00"  "failed"            27       16282       1           71.41                     20362                               "2024-11-26T11:29:03.684327+00:00"
+ 5440  null                                "pending"           27       16357       1           62.64                     20180                               "2024-11-16T17:29:43.540722+00:00"
+ 67    "2024-11-29T04:08:29.746079+00:00"  "cleared"           27       16358       1           86.31                     20179                               "2024-11-28T04:11:31.857431+00:00"
+ 68    "2024-11-11T11:35:40.673632+00:00"  "cleared"           27       16359       1           138.38                    20178                               "2024-11-11T01:19:53.140456+00:00"
+
+(5 rows)
+```
+
+When running this query, Chalk only fetches exactly the base features needed to return `name_match` and `txns`.
+
+Chalk analyzes type annotations to build a directed acyclic graph (DAG) of feature dependencies.
+At inference time, query plans tailored to your input/output schema are generated dynamically by slicing this DAG into sub-graphs.
+This ensures that only the features needed are fetched--optimizing for speed, precision, and cost.
+
+Each node in the DAG is a Chalk expression or resolver (SQL / Python):
+
+- An expression e.g. `is_expensive_purchase: bool = _.amount > 100`
+- A SQL file that connects to an underlying data store like Athena
+- Python resolver i.e. ordinary Python functions (Pandas, httpx, or LLM calls)
```

Comment:
i think the capitalization on these bullets is probably nicer -- could we update the other lists to use the same style?